dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventsExportController.java

```diff
@@ -296,9 +296,40 @@ ResponseEntity<InputStreamResource> getEventDataValueImage(
   }
 
   @GetMapping("/{uid}/changeLog")
-  List<EventChangeLog> getEventByUid(@OpenApi.Param({UID.class, Event.class}) @PathVariable UID uid)
+  Page<ObjectNode> getEventByUid(
+      @OpenApi.Param({UID.class, Event.class}) @PathVariable UID uid,
+      ChangeLogRequestParams requestParams,
+      HttpServletRequest request)
       throws NotFoundException {
 
-    return eventChangeLogService.getEventChangeLog(uid);
+    PageParams pageParams =
+        new PageParams(requestParams.getPage(), requestParams.getPageSize(), false);
+
+    org.hisp.dhis.tracker.export.Page<EventChangeLog> changeLogs =
+        eventChangeLogService.getEventChangeLog(uid, pageParams);
+
+    if (changeLogs.isHasNext()) {
+      changeLogs
+          .getPager()
+          .setNextPage(
+              request.getRequestURI()
+                  + String.format(
+                      "?page=%d&pageSize=%d",
```

Comment:
This is insufficient if we provide non-pagination parameters like `fields`.

If you try 

`https://play.im.dhis2.org/dev/api/organisationUnits?page=2&pageSize=2&fields=displayName`

you see that the query parameters are preserved

```json
{
  "pager": {
    "page": 2,
    "total": 1332,
    "pageSize": 2,
    "nextPage": "https://play.im.dhis2.org/dev/api/organisationUnits?page=3&pageSize=2&fields=displayName",
    "prevPage": "https://play.im.dhis2.org/dev/api/organisationUnits?pageSize=2&fields=displayName",
    "pageCount": 666
  },
  "organisationUnits": [
    {
      "displayName": "Agape CHP"
    },
    {
      "displayName": "Ahamadyya Mission Cl"
    }
  ]
}
```